tests/fake_client.py

```diff
@@ -5,16 +5,182 @@
 import uuid
 from pathlib import Path
 
+import janus
 import socketio
+import numpy as np
+
+from imjoy.helper import dotdict
+from imjoy.workers.worker_template import PluginConnection
+from imjoy.workers.worker3 import JOB_HANDLERS_PY3
 
 logging.basicConfig(stream=sys.stdout)
-_LOGGER = logging.getLogger(__name__)
+logger = logging.getLogger(__name__)
 
-_LOGGER.setLevel(logging.INFO)
+logger.setLevel(logging.INFO)
 
 NAME_SPACE = "/"
 
 
+class ImJoyAPI:
+    """ Represent a set of mock ImJoy API """
+
+    def log(self, message):
+        logger.info("log: %s", message)
+
+    def error(self, message):
+        logger.info("error: %s", message)
+
+    def alert(self, message):
+        logger.info("alert: %s", message)
+
+    def showStatus(self, message):
+        logger.info("showStatus: %s", message)
+
+    def showMessage(self, message):
+        logger.info("showMessage: %s", message)
+
+
+class Plugin:
+    """ Represent a mock proxy plugin """
+
+    def __init__(self, loop, sio, pid, secret):
+        self.conn = None
+        self.loop = loop
+        self.sio = sio
+        self.pid = pid
+        self.secret = secret
+        self._plugin_message_handler = []
+        self.api = None
+        self.imjoy_api = ImJoyAPI()
+        self.janus_queue = janus.Queue(loop=self.loop)
+        self.queue = self.janus_queue.sync_q
+
+        @sio.on("message_from_plugin_" + secret)
+        async def on_message(msg):  # pylint:disable=unused-variable
+            logger.info("Message from plugin: %s", msg)
+            self.message_handler(msg)
+
+    def get_api(self):
+        return self.conn.local["api"]
+
+    async def message_worker(self, async_q, abort=None):
```

Comment:
It would be better to import the function from the production code.

Also make sure there are separate unit tests for the functions that are used to facilitate the tests of other things.

Reply:
The problem here is that the clients are different, one is worker client, the other is a mock browser client. They are similar but not the same.